packages/fxa-settings/src/pages/PostVerify/ThirdPartyAuthCallback/index.tsx

```diff
@@ -3,40 +3,150 @@
+      const state = params.get('state');
+      if (state) {
+        // we may need to deconstruct the state to access/modify the redirect URL
+        const stateParams = new URL(decodeURIComponent(state)).searchParams;
+        const redirect = stateParams.get('redirect_uri');
+        // if the state contains a redirect_uri, we need to redirect to RP
+        // otherwise we redirect internally
+        if (redirect) {
+          hardNavigate(redirect);
+        } else {
+          // general redirect to settings for non-RP
+          // currently, redirect to /settings fails with an "unauthenticated" error from GQL
+          // and redirects to /signin where signin in *again* with ThirdPArty Auth successfully
+          // navigates to /settings
+          // however, after signup we can see the account is successfully created in the db
+          // and added to local storage
+          // need to determine what is missing to directly be considered signed in
+          // when signup with thrid party auth is handled with react from end to end
+          navigate(`/settings${window.location.search}`);
+        }
       }
-    } catch (e) {
-      // noop. navigate to home below.
     }
+  };
 
-    navigate('/', { replace: true });
+  // Persist account data to local storage to match parity with content-server
+  // this allows the recent account to be used for /signin
+  const storeAccountData = (accountData: StoredAccountData) => {
+    persistAccount(accountData);
+    setCurrentAccount(accountData.uid);
+    sessionToken(accountData.sessionToken);
+  };
+
+  async function verifyAndContinue() {
+    const { code, provider } = getAuthParams();
+
+    if (code && provider) {
+      try {
+        // Verify and link the third party account to FxA. Note, this
+        // will create a new FxA account if one does not exist.
+        // The response contains a session token that can be used
+        // to sign the user in to FxA or to complete an Oauth flow.
+        const linkedAccount = await account.verifyAccountThirdParty(
+          code,
+          provider
+        );
+
+        const accountData: StoredAccountData = {
+          // We are using the email that was returned from the Third Party Auth
+          // Not the email entered in the email-first form as they might be different
+          email: linkedAccount.email,
+          uid: linkedAccount.uid,
+          lastLogin: Date.now(),
+          sessionToken: linkedAccount.sessionToken,
+          verified: true,
+          metricsEnabled: true,
+        };
+
+        await storeAccountData(accountData);
+        handlePostSignInNavigation();
+      } catch (error) {
+        // TODO add error handling
+      }
+    } else {
+      // TODO validate what should happen if we hit this page
+      // without the required auth params to verify the account
+      hardNavigateToContentServer('/');
+    }
   }
 
   useEffect(() => {
-    handleOauthResponse();
+    verifyAndContinue();
```

Comment:
Hm, not sure if either function name completely describes what is going on. The `ThirdPartyAuthCallback` page should verify the oauth response from Apple/Google, and then complete the sign-in based on what integration is being used.